### PR TITLE
[CCSpriteCache] allow re-add plist & add testcase

### PR DESCRIPTION
**Pull request by PatriceJiang**
ref: https://github.com/cocos2d/cocos2d-x/pull/19170#discussion_r233737828

### CODE REVIEW

File: cocos/2d/CCSpriteFrameCache.cpp [hunk 1]
```diff
@@ -813,8 +814,9 @@ bool SpriteFrameCache::PlistFramesCache::hasFrame(const std::string &frame) cons
 
 bool SpriteFrameCache::PlistFramesCache::hasPlist(const std::string &plist) const
```
Comment (drelaptop) [inline cocos/2d/CCSpriteFrameCache.cpp]: the impl of `PlistFramesCache::hasPlist` become same with `PlistFramesCache::isPlistUsed`, I think `hasPlist` should be removed

Comment (drelaptop) [reply cocos/2d/CCSpriteFrameCache.cpp]: and what's the meanings of `isPlistUsed `

File: cocos/2d/CCSpriteFrameCache.cpp [hunk 2]
```diff
@@ -370,10 +370,11 @@ void SpriteFrameCache::addSpriteFramesWithFileContent(const std::string& plist_c
 void SpriteFrameCache::addSpriteFramesWithFile(const std::string& plist, const std::string& textureFileName)
 {
     CCASSERT(textureFileName.size()>0, "texture name should not be null");
-    if (_spriteFramesCache.hasPlist(plist))
-    {
-        return; // We already added it
-    }
+    //// allow redundant adding
```
Comment (drelaptop) [inline cocos/2d/CCSpriteFrameCache.cpp]: should be removed, not comment

File: cocos/2d/CCSpriteFrameCache.cpp [hunk 3]
```diff
@@ -424,7 +425,7 @@ void SpriteFrameCache::addSpriteFramesWithFile(const std::string& plist)
             CCLOG("cocos2d: SpriteFrameCache: Trying to use file %s as texture", texturePath.c_str());
         }
         addSpriteFramesWithDictionary(dict, texturePath, plist);
```
Comment (drelaptop) [inline cocos/2d/CCSpriteFrameCache.cpp]: it support redundant adding based on frames? not plist

File: cocos/2d/CCSpriteFrameCache.cpp (hunk 1 above)
Comment (PatriceJiang) [reply cocos/2d/CCSpriteFrameCache.cpp]: isPlistUsed is not used,

File: cocos/2d/CCSpriteFrameCache.cpp [hunk 4]
```diff
@@ -391,45 +382,42 @@ void SpriteFrameCache::addSpriteFramesWithFile(const std::string& plist)
...
-        addSpriteFramesWithDictionary(dict, texturePath, plist);
+        CCLOG("cocos2d: SpriteFrameCache: Trying to use file %s as texture", texturePath.c_str());
     }
+    addSpriteFramesWithDictionary(dict, texturePath, plist);
 }
 
 bool SpriteFrameCache::isSpriteFramesWithFileLoaded(const std::string& plist) const
 {
-    return _spriteFramesCache.hasPlist(plist);
+    return _spriteFramesCache.isPlistUsed(plist);
```
Comment (drelaptop) [inline cocos/2d/CCSpriteFrameCache.cpp]: the meanings of `isSpriteFramesWithFileLoaded ` changed, the origin is
```C++
    /** Check if multiple Sprite Frames from a plist file have been loaded.
    ...
    bool isSpriteFramesWithFileLoaded(const std::string& plist) const;
```
but not it's: Check if __one of__ multiple Sprite Frames from a plist file have been loaded. maybe should change to
```
isPlistUsed && isFullFrames
```

Comment (PatriceJiang) [reply cocos/2d/CCSpriteFrameCache.cpp]: yes, I will add a `full` state.

**Review by drelaptop — APPROVED** [review]

File: tests/cpp-tests/Classes/SpriteFrameCacheTest/SpriteFrameCacheTest.cpp [hunk 5]
```diff
@@ -109,7 +109,6 @@ void SpriteFrameCacheLoadMultipleTimes::loadSpriteFrames(const std::string &file
     SpriteFrame *spriteFrame = SpriteFrameCache::getInstance()->getSpriteFrameByName("grossini.png");
     Texture2D *texture = spriteFrame->getTexture();
     const ssize_t bitsPerKB = 8 * 1024;
```
Comment (drelaptop) [inline tests/cpp-tests/Classes/SpriteFrameCacheTest/SpriteFrameCacheTest.cpp]:
```
Compiling SpriteFrameCacheTest.cpp
❌  /Users/travis/build/cocos2d/cocos2d-x/tests/cpp-tests/Classes/SpriteFrameCacheTest/SpriteFrameCacheTest.cpp:111:19: unused variable 'bitsPerKB' [-Werror,-Wunused-variable]
    const ssize_t bitsPerKB = 8 * 1024;
                                                   ^
▸ Compiling QuestionContainerSprite.cpp
** BUILD FAILED **
```

**Pull request merged**